examples/react-fcl-demo/src/demo/DeployContract.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {useState} from "react"
+
+import * as fcl from "@onflow/fcl"
+import * as t from "@onflow/types"
+
+import Card from '../components/Card'
+import Header from '../components/Header'
+import Code from '../components/Code'  
+
+const deployTransaction = `\
+transaction(code: String) {
+  prepare(acct: AuthAccount) {
+    acct.setCode(code.decodeHex())
+  }
+}
+`
+
+const simpleContract = `
+pub contract HelloWorld {
+  pub let greeting: String
+  pub event HelloEvent(message: String)
+
+  init() {
+    self.greeting = "Hello, World!"
+  }
+
+  pub fun hello(message: String): String {
+    emit HelloEvent(message: message)
+    return self.greeting
+  }
+}
+`
+
+const DeployContract = () => {
+  const [status, setStatus] = useState("Not started")
+  const [transaction, setTransaction] = useState(null)
+
+  const runTransaction = async (event: any) => {
+    event.preventDefault()
+    
+    setStatus("Resolving...")
+
+    const blockResponse = await fcl.send([
+      fcl.getLatestBlock(),
+    ])
+
+    const block = await fcl.decode(blockResponse)
+    
+    try {
+      const { transactionId } = await fcl.send([
+        fcl.transaction(deployTransaction),
+        fcl.args([
+          fcl.arg(
+            Buffer.from(simpleContract, "utf8").toString("hex"),
+            t.String
+          )
+        ]),
+        fcl.proposer(fcl.currentUser().authorization),
+        fcl.authorizations([
+          fcl.currentUser().authorization
+        ]),
+        fcl.payer(fcl.currentUser().authorization),
+        fcl.ref(block.id),
+      ])
+
+      setStatus("Transaction sent, waiting for confirmation")
+
+      const unsub = fcl
+        .tx({ transactionId })
+        .subscribe((transaction: any) => {
+          setTransaction(transaction)
+          
+          if (fcl.tx.isSealed(transaction)) {
+            setStatus("Transaction is Sealed")
+            unsub()
+          }
+        })
```

Comment:
In 99% of cases you only need to wait for execution. `fcl.tx.isExecuted(txStatus)`.

This is super common though so we tried to make this easier. `fcl.tx(txId).onceExecuted()` is a promise that will resolve when the transaction has executed (or sealed) and will reject if there are any errors. It resolves and rejects with the raw `txStatus` object (the same one you are currently passing into `fcl.tx.isExecuted(txStatus)`).

```javascript
var resp = await fcl.send([...])

try {
  var txStatus = await fcl.tx(resp).onceExecuted()
  setStatus("Transaction Executed")
} catch (error) {
  // error says why it failed...
  setStatus("Transaction Executed but failed")
}
```

Interesting note. the `txStatus` also includes any events that were emitted during the transaction.

Reply:
fixed